src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -712,6 +712,11 @@ public boolean process(Person person, long time) {
       }
 
       if (value != null) {
+        // the module is setting an attribute to be the value of an existing attribute
+        if (person.attributes.containsKey(value)) {
```

Comment:
This is an interesting idea but scares me a little because of the possibility of false positives, if there's an unintentional overlap between the name of 1 attribute and the value of another. Is there a way we could detect that issue in advance? Offhand I can't think of anything because we can't know the intention of the author.

Reply:
Theoretically, we could add another attribute to the `SetAttribute` class called `valueAttribute` or something, and modify the Module Builder to have additional fields (perhaps a checkbox or something else to identify it as a reference to another attribute rather than a string).

Theoretically, we could also detect if the string value a user entered in the UI was the name of another attribute, so long as that attribute was declared within this module or is part of the precomputed `module-builder/src/data/modules.js` file.

Reply:
That being said, I would struggle implementing those features in the module builder.

Reply:
I made the change in the code for a `valueAttribute` but I am leaving the module builder PR for later, see https://github.com/synthetichealth/module-builder/issues/302